include/stdexec/execution.hpp

```diff
@@ -5485,6 +5485,203 @@ namespace stdexec {
+          __ops_;
+
+      using __result_type = __result_type_t<env_of_t<_Receiver>, _Senders...>;
+      __result_type __result_;
+
+      template <class _CPO, class... _Args>
+      void notify(_CPO, _Args&&... __args) noexcept {
+        bool __expect = false;
+        if (__emplaced_.compare_exchange_strong(__expect, true, std::memory_order_relaxed,
+                                                std::memory_order_relaxed)) {
+          // This emplacement can happen only once
+          __result_.template emplace<std::tuple<_CPO, std::decay_t<_Args>...>>(
+              _CPO{}, ((_Args &&) __args)...);
+          // stop pending operations
+          __stop_source_.request_stop();
+        }
+        // make __result_ emplacement visible when __count_ goes from one to zero
+        // This relies on the fact that each sender will call notify() at most once
+        if (__count_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
+          __on_stop_.reset();
+          std::visit(
+              [this]<class _Tuple>(_Tuple&& __result) {
+                if constexpr (std::same_as<std::decay_t<_Tuple>, std::monostate>) {
+                  set_stopped((_Receiver &&) __receiver_);
+                } else {
+                  auto stop_token = get_stop_token(get_env(__receiver_));
+                  if (stop_token.stop_requested()) {
+                    set_stopped((_Receiver &&) __receiver_);
+                    return;
+                  }
+                  std::apply(
+                      [this]<class _C, class... _As>(_C, _As&&... __args) noexcept {
+                        _C{}((_Receiver &&) __receiver_, (_As &&) __args...);
+                      },
+                      (_Tuple &&) __result);
+                }
+              },
+              (__result_type &&) __result_);
+        }
+      }
+
+      friend void tag_invoke(start_t, __t& __self) noexcept {
+        __self.__on_stop_.emplace(get_stop_token(get_env(__self.__receiver_)),
+                                  __on_stop_requested{__self.__stop_source_});
+        if (__self.__stop_source_.stop_requested()) {
+          stdexec::set_stopped((_Receiver &&) __self.__receiver_);
+        } else {
+          std::apply([](auto&... __ops) { (start(__ops), ...); }, __self.__ops_);
+        }
+      }
+    };
+  };
+  template <class _Receiver, class... _Senders>
+  using __op = __t<__op_id<_Receiver, _Senders...>>;
+
+  template <class _Sender, class... _Senders> struct __sender_id {
+    struct __t {
+      using __id = __sender_id;
+
+      std::tuple<_Sender, _Senders...> __senders_;
+
+      template <class _R>
+      friend __op<std::remove_cvref_t<_R>, _Sender, _Senders...> 
+        tag_invoke(connect_t, __t&& __self, _R&& __rcvr) noexcept {
+          return {(std::tuple<_Sender, _Senders...> &&) __self.__senders_, (_R &&) __rcvr};
+        }
+
+      friend __empty_env tag_invoke(get_env_t, const __t& __self) noexcept {
+        return {};
+      }
```

Comment:
This is the default behavior of `get_env` now, so this friend function isn't needed. I would strike it.